### DISCUSSION TITLE
Artwork no longer works after reinstalling foobar...

### DISCUSSION THREAD

**Discussion by smokeanthrax** [s0hv/foo_discord_rich#8] [Q&A]
hi,

apologies for the lengthy post. long story short, i had to reinstall foobar a couple hours ago (after fully uninstalling first) and artwork no longer works. if you need more details, read on

back in December i followed [this tutorial](https://www.youtube.com/watch?v=E-uq0WT2pwk). the plugin worked and has worked ever since with zero issues. it continued to work when i updated from v1.6 to v2.0 (beta 26), although i kept all my configurations when updating. i should note i've never really done much with foobar and just used Default UI with this component, Facets and a couple others.

fast forward to today, i ran into some huge issues with another component (all totally unrelated to this plugin) and had to do a clean reinstall of foobar. i installed beta 29, and went to set up this plugin again, exactly as i'd done before — i installed the component (same version, 1.3.0-beta), ticked all the same boxes, put in the same artwork upload command (upload.py and my Python installation were untouched), identifier key, discord app token, everything.

no dice. artwork does not display.

i've tried a myriad of different things and i'm completely at a loss... i tried going back to v1.6, v2.0 beta 26, neither worked. i tried a portable installation of beta 29, beta 26 and v1.6 and it didn't work. i tried creating a new discord application, registering a new imgur application, using [imgur-uploader.exe](https://github.com/s0hv/rust-imgur-upload/releases/tag/0.2.0) instead of the whole upload.py thing, using [this script](https://gist.github.com/s0hv/5c07cfb4b939ee619d0efcc047991ceb) instead of the old one in that YouTube video, using quotation marks around the file paths and **nothing worked**. none of it worked on any of the different versions i tried, portable or standard

in the console, it seems to be uploading artwork just fine. yet it doesn't show on discord. this is what shows up when i play a track: 

`Opening track for playback: "E:\Music\Madvillain (MF DOOM & Madlib) - Madvillainy (2004) [WEB-FLAC]\08 Rainbows.flac"
Wave cache: finished analysis of "E:\Music\Madvillain (MF DOOM & Madlib) - Madvillainy (2004) [WEB-FLAC]\08 Rainbows.flac"`

when i clear and generate an artwork url:

`Discord Rich Presence Integration v1.3.0-beta-2+b5997cc: 1 entries cleared
Discord Rich Presence Integration v1.3.0-beta-2+b5997cc: Found url https://i.imgur.com/wp1waTq.jpg with image hash 68416D4621737424FBD44082E5457723
Discord Rich Presence Integration v1.3.0-beta-2+b5997cc: 1 entries updated`

i genuinely have no idea what changed. i hadn't even configured my first installation of foobar differently at all... i followed the YouTube tutorial to a tee and it didn't work, i followed your documentation to a tee and it didn't work either.

i would really appreciate any help

**Comment by s0hv** [comment]
Looks like the upload part works fine as that imgur link is valid. Does the extension work at all? Meaning that does anything show up in your discord status when you start playing.

**Comment by smokeanthrax** [comment]
So the artwork magically started working. I don’t know what the issue was

But yeah, the plugin was working otherwise. It displayed my music, the
foobar logo, and pause/play symbols on discord just fine

**Comment by s0hv** [comment]
Might've been something on discords side then as the images go through discords media proxy
